### PR TITLE
Add base files for flask image with flake8 and autopep8

### LINKED ISSUE #12: Flask on Docker proof of concept
Produce proof of concept code showing that a Flask API can be developed and run in a Docker container.

### PR DESCRIPTION
**Pull request by zdc217**
+ Add Dockerfile
+ Add Makefile
+ Add requirements file
+ Add basic app file with canary endpoint
+ Update readme

### CODE REVIEW

**Review by MKletz — APPROVED** [review]

File: requirements/requirements.in [hunk 1]
```diff
@@ -0,0 +1,3 @@
+flask
+flake8
```
Comment (ddriddle) [inline requirements/requirements.in]: I recommend moving the `flake8` and `autopep8` deps into a separate `requirements-dev.txt` file. I also recommend you move the `requirements` files to the top-level. I have never seen a `requirements` directory in use before. It is a non-standard practice.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+.PHONY: build run clean shell
```
Comment (ddriddle) [inline Makefile]: Add `all`.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+all: requirements.txt run
+
+build: .build
+.build: $(SRCS)
+	docker build -t $(IMAGE) .
+	echo > $@
+
+run: build
+	docker run $(DARGS) -p $(PORT) $(IMAGE)
+
```
Comment (ddriddle) [inline Makefile:20]: Add a `test` that runs `pep8` and future unit tests.

**Review by ddriddle — CHANGES_REQUESTED** [review]
@zcarrington After you have the `make test` feature added I also recommend you enable Drone, and add a `.drone.yml` file with the following contents:

```
pipeline:
  build:
    image: python:3.8
    commands:
      - make deps
      - make test
```

Note that `make deps` should install all requirements needed for testing and running the code. I recommend the following be added to the Makefile:

```
deps:
        pip install -r requirements.txt
        pip install -r requirements-dev.txt
```

**Comment by ddriddle** [comment]
@zdc to enable Drone for this repo email: techservices-git@illinois.edu.

**Comment by ddriddle** [comment]
@zdc you can check if a Drone build succeeds by checking: 

https://drone.techservices.illinois.edu/techservicesillinois/secdev-quarantinenetwork

You can also add a markdown badge your README.md:

```markdown
[![Build Status](https://drone.techservices.illinois.edu/api/badges/techservicesillinois/secdev-quarantinenetwork/status.svg)](https://drone.techservices.illinois.edu/techservicesillinois/secdev-quarantinenetwork)
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,56 @@
-# secdev-quarantinenetwork
-source code for the Quarantine Network self-service portal
+# SecDev Quarantine Network
+
```
Comment (ddriddle) [inline README.md:2]: Please add a badge here for Drone:

```
[![Build Status](https://drone.techservices.illinois.edu/api/badges/techservicesillinois/secdev-quarantinenetwork/status.svg)](https://drone.techservices.illinois.edu/techservicesillinois/secdev-quarantinenetwork)
```

Please note that this points to the `master` branch.

**Review by edthedev — APPROVED** [review]
Approved - pending the changes requested by David.

**Review by ddriddle — APPROVED** [review]

**Review by tzturner — APPROVED** [review]

**Pull request merged**